### ISSUE TITLE
Issue: Apache Validation Error, AH00526: Syntax error on line 1

### ISSUE THREAD

**Issue by laura-outgang** [custom-dev-tools/wampserver-ssl-auto-config#10]
Having this issue, any help is greatly appreciated. Thanks!

###  WampServer Apache Validation Error

> WampServer SSL Auto Config (v1.2.3)
>   -----------------------------------
> 
>   11:17:52.56 : Parsed configuration file.
>   11:17:52.57 : Common 'logs' directory already exists.
>   11:17:52.58 : Script initialised.
>   11:17:52.60 : ---------------------------------------------
>   11:17:52.61 : Found WampServer installation path.
>   11:17:52.63 : Found WampServer Apache path.
>   11:17:52.64 : Found WampServer Apache installation(s):
>   11:17:52.66 :   'apache2.4.54.2'
>   11:17:52.68 : Found OS 'hosts' file.
>   11:17:52.70 : Found OS Apache service name.
>   11:17:52.71 : Validating Apache config file(s):
> 
>   -----------------------------------------------------------
> 
>                             FAILURE
> 
>   -----------------------------------------------------------
> 
>   WampServer Apache Validation Error
> 
>   File:  See below
>   Error: AH00526: Syntax error on line 1 of C:/Outgang/Web/Tools/ssl/vhosts/https/conf/httpd-ssl.conf:
> 
>   Please correct the error in the stated configuration file and restart WampServer.
> 
>   -----------------------------------------------------------
> 
>   Press any key to exit.

### **To Reproduce**
Having this issue when running ssl_config.bat config.ini from the command line. It's not the first time I'm installing an ssl certificate. I uninstalled wampserver yesterday and reinstalled it with a new version (3.3.0) and trying to get my local development server to be running again. 

### Versions
WAMPserver: 3.3.0
SSL auto config: 1.2.3
Apache 2.4.54.2
Win 10. 

### **Additional context**

> config.ini:
> ;--------------------------;
> ; WampServer Configuration ;
> ;--------------------------;
> 
> ; Your WampServer installation path.
> wampServerInstallPath=C:\wamp64
> 
> ; Your custom path to store your SSL certificates, keys, logs and vhost files.
> ; Ensure this path is NOT within your WampServer installation path.
> wampServerExtensionsPath=C:\Outgang\Web\Tools\ssl
> 
> ;-------------------------;
> ; SSL Certificate Details ;
> ;-------------------------;
> 
> ; These (common) ssl certificate details are used to build each developments domain name certificate.
> ;
> ; sslCity:             The full name of a city.
> ; sslState:            The full name of a state.
> ; sslCountry:          The two letter ISO code of a country.
> ; sslOrganisation:     The organisation name.
> ; sslOrganisationUnit: The unit name of a organisation.
> ; sslEmail:            Use the 'local' part of an email address followed by the @ (at) symbol only.
> ;                      IMPORTANT: Do not include the 'domain' part of the email address as the hostname will be auto-appended.
> ; sslDays:             The number of days you would like the certificates to remain valid for.
> 
> sslCity=Montreal
> sslState=Quebec
> sslCountry=CA
> sslOrganization=Outgang
> sslOrganizationUnit=Studio
> sslEmail=laura@
> sslDays=3650
> 
> ;---------------------;
> ; Development Domains ;
> ;---------------------;
> 
> [outgang.studio]
> hostname=outgang.studio
> documentRoot=C:\wamp64\www\outgang.studio\public_html

### Log file: 

> 
> ===========================================================
> 
> 2023-10-14 : WampServer SSL Auto Config Script (v1.2.3)
> 
> 10:58:57.31 : Script initialised. 
> 10:58:57.33 : --------------------------------------------- 
> 10:58:57.33 : Found WampServer installation path at 'C:\wamp64' 
> 10:58:57.35 : Found WampServer Apache path at 'C:\wamp64\bin\apache' 
> 10:58:57.37 : Found WampServer Apache installation(s): 
> 10:58:57.39 :   'apache2.4.54.2' 
> 10:58:57.40 : Found OS 'hosts' file at 'C:\WINDOWS\System32\drivers\etc\hosts' 
> 10:58:57.42 : Found OS Apache service name 'wampapache64' 
> 10:58:57.44 : Validating Apache config file(s): 
> 
> ===========================================================
> 
> 2023-10-14 : WampServer SSL Auto Config Script (v1.2.3)
> 
> 11:02:05.13 : Script initialised. 
> 11:02:05.15 : --------------------------------------------- 
> 11:02:05.16 : Found WampServer installation path at 'C:\wamp64' 
> 11:02:05.17 : Found WampServer Apache path at 'C:\wamp64\bin\apache' 
> 11:02:05.19 : Found WampServer Apache installation(s): 
> 11:02:05.21 :   'apache2.4.54.2' 
> 11:02:05.22 : Found OS 'hosts' file at 'C:\WINDOWS\System32\drivers\etc\hosts' 
> 11:02:05.25 : Found OS Apache service name 'wampapache64' 
> 11:02:05.27 : Validating Apache config file(s): 
> 
> ===========================================================
> 
> 2023-10-14 : WampServer SSL Auto Config Script (v1.2.3)
> 
> 11:07:41.05 : Script initialised. 
> 11:07:41.07 : --------------------------------------------- 
> 11:07:41.08 : Found WampServer installation path at 'C:\wamp64' 
> 11:07:41.09 : Found WampServer Apache path at 'C:\wamp64\bin\apache' 
> 11:07:41.11 : Found WampServer Apache installation(s): 
> 11:07:41.14 :   'apache2.4.54.2' 
> 11:07:41.15 : Found OS 'hosts' file at 'C:\WINDOWS\System32\drivers\etc\hosts' 
> 11:07:41.17 : Found OS Apache service name 'wampapache64' 
> 11:07:41.19 : Validating Apache config file(s): 
> 11:07:41.44 :   'apache2.4.54.2' 
> 11:07:41.46 : Using 'openssl.exe' from 'apache2.4.54.2' 
> 11:07:41.51 : Starting Apache service. 
> 
> ===========================================================
> 
> 2023-10-14 : WampServer SSL Auto Config Script (v1.2.3)
> 
> 11:08:37.64 : Script initialised. 
> 11:08:37.65 : --------------------------------------------- 
> 11:08:37.66 : Found WampServer installation path at 'C:\wamp64' 
> 11:08:37.67 : Found WampServer Apache path at 'C:\wamp64\bin\apache' 
> 11:08:37.69 : Found WampServer Apache installation(s): 
> 11:08:37.72 :   'apache2.4.54.2' 
> 11:08:37.72 : Found OS 'hosts' file at 'C:\WINDOWS\System32\drivers\etc\hosts' 
> 11:08:37.75 : Found OS Apache service name 'wampapache64' 
> 11:08:37.77 : Validating Apache config file(s): 
> 11:08:38.01 :   'apache2.4.54.2' 
> 11:08:38.03 : Using 'openssl.exe' from 'apache2.4.54.2' 
> 11:08:38.06 : Apache service already started. 
> 11:08:38.08 : OS 'hosts' file already backed up. 
> 11:08:38.09 : Backing up Apache 'httpd.conf' file(s): 
> 11:08:38.11 :   'apache2.4.54.2' already backed up. 
> 11:08:38.11 : Created common 'certs' directory at 'C:\Outgang\Web\Tools\ssl\certs' 
> 11:08:38.12 : Created common 'vhosts\http' directory at 'C:\Outgang\Web\Tools\ssl\vhosts\http' 
> 11:08:38.13 : Created common 'vhosts\https' directory at 'C:\Outgang\Web\Tools\ssl\vhosts\https' 
> 11:08:38.15 : Created common 'httpd-ssl.conf' file. 
> 11:08:38.17 : --------------------------------------------- 
> 11:08:38.18 : outgang.studio (outgang.studio) 
> 11:08:38.20 :   'certs' directory created. 
> 11:08:38.21 :   'logs' directory created. 
> 11:08:38.24 :   Created 'openssl.conf' file. 
> 11:08:38.26 :   Created Virtual Host http file. 
> 11:08:38.28 :   Created Virtual Host https file. 
> 11:08:38.33 :   Deleted old certificate from store. 
> 11:08:38.40 :     Generated certificate keys. 
> 11:08:38.43 :     Removed certificate passphrase. 
> 11:08:38.46 :     Generated Certificate. 
> 11:08:38.48 :     Deleted 'private.pem' file. 
> 11:08:38.52 :   Added new certificate to store. 
> 11:08:38.55 :   Attempting to add hostname to OS 'hosts' file. 
> 11:08:38.58 :   Unable to added hostname to OS 'hosts' file. 
> 11:08:38.60 : --------------------------------------------- 
> 11:08:38.62 : Updating Apache primary config file(s): 
> 11:08:38.63 :   'apache2.4.54.2\conf\httpd.conf' 
> 11:08:38.69 :     'socache_shmcb_module' already uncommented. 
> 11:08:38.74 :     'ssl_module' already uncommented. 
> 11:08:38.79 :     'http2_module' already uncommented. 
> 11:08:38.83 :     Added SSL Config link. 
> 11:08:38.88 :     Added HTTP vhosts link. 
> 11:08:38.93 :     Added HTTPS vhosts link. 
> 
> ===========================================================
> 
> 2023-10-14 : WampServer SSL Auto Config Script (v1.2.3)
> 
> 11:17:52.59 : Script initialised. 
> 11:17:52.60 : --------------------------------------------- 
> 11:17:52.61 : Found WampServer installation path at 'C:\wamp64' 
> 11:17:52.62 : Found WampServer Apache path at 'C:\wamp64\bin\apache' 
> 11:17:52.64 : Found WampServer Apache installation(s): 
> 11:17:52.67 :   'apache2.4.54.2' 
> 11:17:52.67 : Found OS 'hosts' file at 'C:\WINDOWS\System32\drivers\etc\hosts' 
> 11:17:52.70 : Found OS Apache service name 'wampapache64' 
> 11:17:52.72 : Validating Apache config file(s): 
> 
> ===========================================================
> 
> 2023-10-14 : WampServer SSL Auto Config Script (v1.2.3)
> 
> 11:34:23.64 : Script initialised. 
> 11:34:23.66 : --------------------------------------------- 
> 11:34:23.67 : Found WampServer installation path at 'C:\wamp64' 
> 11:34:23.68 : Found WampServer Apache path at 'C:\wamp64\bin\apache' 
> 11:34:23.71 : Found WampServer Apache installation(s): 
> 11:34:23.73 :   'apache2.4.54.2' 
> 11:34:23.74 : Found OS 'hosts' file at 'C:\WINDOWS\System32\drivers\etc\hosts' 
> 11:34:23.77 : Found OS Apache service name 'wampapache64' 
> 11:34:23.79 : Validating Apache config file(s): 
> 11:34:24.04 :   'apache2.4.54.2' 
> 11:34:24.06 : Using 'openssl.exe' from 'apache2.4.54.2' 
> 11:34:24.10 : Apache service already started. 
> 11:34:24.12 : OS 'hosts' file already backed up. 
> 11:34:24.14 : Backing up Apache 'httpd.conf' file(s): 
> 11:34:24.15 :   'apache2.4.54.2' already backed up. 
> 11:34:24.16 : Common 'certs' directory already exists at 'C:\Outgang\Web\Tools\ssl\certs' 
> 11:34:24.17 : Common 'vhosts\http' directory already exists at 'C:\Outgang\Web\Tools\ssl\vhosts\http' 
> 11:34:24.17 : Common 'vhosts\https' directory already exists at 'C:\Outgang\Web\Tools\ssl\vhosts\https' 
> 11:34:24.20 : Re-created common 'httpd-ssl.conf' file. 
> 11:34:24.22 : --------------------------------------------- 
> 11:34:24.23 : outgang.studio (outgang.studio) 
> 11:34:24.25 :   'certs' directory already exists. 
> 11:34:24.27 :   'logs' directory already exists. 
> 11:34:24.29 :   Re-created 'openssl.cnf' file. 
> 11:34:24.32 :   Re-created Virtual Host http file. 
> 11:34:24.35 :   Re-created Virtual Host https file. 
> 11:34:24.40 :   Deleted old certificate from store. 
> 11:34:24.52 :     Generated certificate keys. 
> 11:34:24.55 :     Removed certificate passphrase. 
> 11:34:24.59 :     Generated Certificate. 
> 11:34:24.61 :     Deleted 'private.pem' file. 
> 11:34:24.65 :   Added new certificate to store. 
> 11:34:24.69 :   Attempting to add hostname to OS 'hosts' file. 
> 11:34:24.72 :   Unable to added hostname to OS 'hosts' file. 
> 11:34:24.74 : --------------------------------------------- 
> 11:34:24.75 : Updating Apache primary config file(s): 
> 11:34:24.77 :   'apache2.4.54.2\conf\httpd.conf' 
> 11:34:24.82 :     'socache_shmcb_module' already uncommented. 
> 11:34:24.88 :     'ssl_module' already uncommented. 
> 11:34:24.93 :     'http2_module' already uncommented. 
> 11:34:24.98 :     SSL Config link already added. 
> 11:34:25.03 :     HTTP vhosts link already added. 
> 11:34:25.07 :     HTTPS vhosts link already added. 
> 
> ===========================================================
> 
> 2023-10-14 : WampServer SSL Auto Config Script (v1.2.3)
> 
> 11:54:25.42 : Script initialised. 
> 11:54:25.44 : --------------------------------------------- 
> 11:54:25.44 : Found WampServer installation path at 'C:\wamp64' 
> 11:54:25.46 : Found WampServer Apache path at 'C:\wamp64\bin\apache' 
> 11:54:25.48 : Found WampServer Apache installation(s): 
> 11:54:25.50 :   'apache2.4.54.2' 
> 11:54:25.51 : Found OS 'hosts' file at 'C:\WINDOWS\System32\drivers\etc\hosts' 
> 11:54:25.53 : Found OS Apache service name 'wampapache64' 
> 11:54:25.56 : Validating Apache config file(s): 

Thanks for the help!

**Comment by laura-outgang** [comment]
Contents of the conf file: 

> Listen 443
> 
> #   SSL Cipher Suite:
> SSLCipherSuite HIGH:MEDIUM:!MD5:!RC4:!3DES
> SSLProxyCipherSuite HIGH:MEDIUM:!MD5:!RC4:!3DES
> 
> #  Enforce the server's cipher order.
> SSLHonorCipherOrder on
> 
> #   SSL Protocol support:
> SSLProtocol all -SSLv3
> SSLProxyProtocol all -SSLv3
> 
> #   Pass Phrase Dialog:
> SSLPassPhraseDialog  builtin
> 
> #   Inter-Process Session Cache:
> SSLSessionCache        "shmcb:c:/Apache24/logs/ssl_scache(512000)"
> SSLSessionCacheTimeout  300

**Comment by laura-outgang** [comment]
Turns out there were two issues. 
The first is that I wasn't running cmd as an admin (which is a duh), visible from the fact that the script couldn't modify the win hosts file, but that became apparent after the Apache syntax issue was resolved. 

After having the issue detailed above, I couldn't start Apache again, at all. Apache was complaining of duplicate statements referring to listening to port 443. I tracked the other 'Listen 443' statement to /apache[version]/conf/extra/httpd-ssl.conf at line 36. I commented that statement and the other duplicate statements from the conf file above. Apache restarted and I finally was able to finish the SSL auto config installation (after running cmd in admin mode). 

I have no clue if this is a proper fix, I highly doubt it, and I'm still wondering how the issue came to be in the first place, but when I reinstalled Wamp yesterday I was using SSL auto config v1.1 and ran into the issue here: https://github.com/custom-dev-tools/wampserver-ssl-auto-config/issues/5#issue-1192627005. I tried a few other solutions online, none of which worked, which is probably how I winded up where I am right now. In all cases, it works, so I'm happy, and hopefully this will be useful for anyone else with the same issue.
